narwhals/_spark_like/expr.py

```diff
@@ -788,10 +821,7 @@ def rolling_std(
     def rank(self, method: RankMethod, *, descending: bool) -> Self:
         func_name = self._REMAP_RANK_METHOD[method]
 
-        def _rank(_input: Column) -> Column:
-            order_by = self._sort(_input, descending=descending, nulls_last=True)
-            window = self.partition_by().orderBy(*order_by)
-            count_window = self.partition_by(_input)
+        def _rank(_input: Column, window: WindowSpec, count_window: WindowSpec) -> Column:
```

Comment:
I think the way you've done it in `_duckdb` is nice, where you share `order_by` between the two functions

Could we do that here too, so that the signatures are aligned?

Reply:
Thank you @MarcoGorelli ! Yes makes sense, I aligned them now